File: src/j4g/kotlinmaiden/command/Greet.java
package j4g.kotlinmaiden.command;

import j4g.kotlinmaiden.component.Command;
import j4g.kotlinmaiden.component.Help;
import j4g.kotlinmaiden.component.Name;

@Name("greet")
@Help("greet <name> - Greets a person")
public class Greet implements Command {
    @Override
    public boolean test(String[] args) {
        return args.length > 1;
    }

    @Override
    public void execute(String[] args) {
        System.out.println("Hello " + args[1]);
    }
}

File: src/j4g/kotlinmaiden/command/Greet.kt
package j4g.kotlinmaiden.command

import j4g.kotlinmaiden.component.*
import j4g.kotlinmaiden.component.Help

@Name("greet")
@Help("greet <name> - Greets a person")
class Greet : Command {
    override fun test(args: Array<String>): Boolean {
        return args.size > 1
    }

    override fun execute(args: Array<String>) {
        println("Hello " + args[1])
    }
}